pkg/ccl/changefeedccl/changefeed_stmt.go

```diff
@@ -1223,3 +1213,99 @@ func failureTypeForStartupError(err error) changefeedbase.FailureType {
 	}
 	return changefeedbase.OnStartup
 }
+
+// checkPrivileges iterates over all the descriptors returned by ResolveTargetsToDescriptors,
+// which includes descriptors for each targeted table and each database containing one or more
+// targeted tables, mixed together in an arbitrary order. We want to honor privileges at either
+// the database level or the table level, so when we hit a missing privilege at the table level
+// we don't know whether that's fatal until we process its parent descriptor.
+func checkPrivileges(
+	ctx context.Context, p singlePrivilegeChecker, descriptors []catalog.Descriptor,
+) error {
```

Comment:
Should we just use resolver.ResolveExisting and call checkPrivilege for users?
Alternatively, the same backup package exposts NewResolver method -- that one seems pretty easy to use